src/applications/accreditation/21a/pages/02-military-service-chapter/militaryServicePages.js

```diff
@@ -75,44 +77,87 @@ const introPage = {
   },
 };
 
+const isCurrentlyServing = (formData, index) => {
+  return formData?.militaryServiceExperiences?.[index]?.currentlyServing;
+};
+
 /** @returns {PageSchema} */
-const militaryServiceExperiencePage = {
+const branchAndDateRangePage = {
   uiSchema: {
-    ...titleUI(
-      'Military service experience',
-      'You will have the option of adding additional periods of service on the next page.',
-    ),
-    serviceBranch: selectUI('Branch of service'),
-    serviceDateRange: currentOrPastDateRangeUI(
+    ...arrayBuilderItemFirstPageTitleUI({
+      title: 'Military service experience',
+      description:
+        'List all periods of military service experience. You will be able to add additional periods of service on subsequent screens.',
+      nounSingular: arrayBuilderOptions.nounSingular,
+    }),
+    branch: selectUI('Branch of service'),
+    dateRange: currentOrPastDateRangeUI(
+      { title: 'Service start date' },
       {
-        title: 'Active service start date',
+        title: 'Service end date',
+        hideIf: (formData, index) => isCurrentlyServing(formData, index),
```

Comment:
really like this approach as well. nice!

![Screenshot 2024-08-16 at 11 28 14 AM](https://github.com/user-attachments/assets/69796200-b1aa-4a0d-ba19-ff936e5af4a0)
![Screenshot 2024-08-16 at 11 28 17 AM](https://github.com/user-attachments/assets/728d0969-e23e-4bf1-ab1b-7f1fb4f9230c)